packages/fxa-settings/src/components/Settings/FlowRecoveryKeyHint/index.test.tsx

```diff
@@ -0,0 +1,147 @@
+
+const mockAccount = () => {
+  return account;
+};
+
+const navigateForward = jest.fn();
+const navigateBackward = jest.fn();
+const localizedPageTitle = 'Account Recovery Key';
+const localizedBackButtonTitle = 'Back to settings';
+
+jest.mock('../../../lib/metrics', () => ({
+  usePageViewEvent: jest.fn(),
+  logViewEvent: jest.fn(),
+}));
+
+jest.mock('../../../models', () => ({
+  ...jest.requireActual('../../../models'),
+  useAccount: () => mockAccount(),
+}));
+
+const renderWithProps = () => {
+  renderWithRouter(
+    <FlowRecoveryKeyHint
+      {...{
+        localizedBackButtonTitle,
+        localizedPageTitle,
+        navigateForward,
+        navigateBackward,
+        viewName,
+      }}
+    />
+  );
+};
+
+describe('FlowRecoveryKeyHint', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders as expected', async () => {
+    renderWithProps();
+
+    screen.getByRole('heading', { level: 1, name: 'Account Recovery Key' });
+    screen.getByRole('progressbar', { name: 'Step 4 of 4.' });
+    screen.getByRole('heading', {
+      level: 2,
+      name: 'Help yourself find your key later',
+    });
+    screen.getByText(
+      'Remember where you saved your key by creating a hint. If you forget your password and canʼt find your key, you might not be able to get your data back.'
+    );
+    screen.getByRole('textbox', { name: 'Enter a hint (optional)' });
+    screen.getByRole('button', { name: 'Finish' });
+  });
+
+  it('emits the expected metrics when the user lands on this step of the flow', () => {
+    renderWithProps();
+    expect(logViewEvent).toBeCalledTimes(1);
+    expect(logViewEvent).toBeCalledWith(viewName, 'hint-step-view');
+  });
+
+  it('saves the hint on submit if the user has entered a valid hint in the text input', async () => {
+    renderWithProps();
+    const hintValue = 'Ye Olde Hint';
+    const textInput = screen.getByRole('textbox', {
+      name: 'Enter a hint (optional)',
+    });
+    const submitButton = screen.getByText('Finish');
+    fireEvent.input(textInput, {
+      target: { value: hintValue },
+    });
+    await waitFor(() => {
+      expect(textInput).toHaveValue(hintValue);
+    });
+    fireEvent.click(submitButton);
+    await waitFor(() => {
+      expect(logViewEvent).toBeCalledWith(viewName, 'hint-submit');
+      expect(account.updateRecoveryKeyHint).toBeCalledWith(hintValue);
+      expect(logViewEvent).toBeCalledWith(viewName, 'hint-success');
+      expect(navigateForward).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  it('displays error tooltip if the hint is too long', async () => {
+    renderWithProps();
+    const hintValueTooLong =
+      'eunqanrhdfiwnzrrigazwjkxywdupiwqlzkneaniwxcnaltdbwlbbtuyozbyxkmqvoiaehcpycjvejabhhyxnvcauxeokprdbbebgmizsphjlbtsscokweckspeqndegwzmlhyrmglbnkmdwvynitqckyubpchplpsxevwtczeuujkznpvtjrchkpbngnmlkqwinzxwswdwinyrgvkvusrpkqcitwznqcsaajqmxtqjlpxxfqbemtxsvolaygntkyvzdtafltvbvvtrxgqfnvdktleuhflkcmwbfracrpbos';
```

Comment:
👍 this is a great idea

Reply:
Less interesting input, but you could also use something like `'a'.repeat(maxLength + 5)` (although, you would need to export this from index.tsx: `const maxLength: number = 255;`).